python-evaluator-transform/README.rst

```diff
@@ -0,0 +1,99 @@
+================
+Python Evaluator
+================
+
+Plugin to use Python code as an ETL Transform.
+
+Sources: Batch: Apache Cassandra
+================================
+
+Description
+-----------
+Executes user-provided Python code that transforms one record into another.
+Input records are converted into Python dictionaries which can be directly accessed
+in Python code.
+
+Use Case
+--------
+The Script transform is used when other transforms cannot meet your needs.
+For example, you may want to multiply a field by 1024 and rename it from ``'gigabytes'``
+to ``'megabytes'``. Or you might want to convert a timestamp into a human-readable date string.
+
+Properties
+----------
+**script:** Python code defining how to transform one record into another. The script must
+implement a function called ``'transform'``, which takes as input a Python dictionary (representing
+the input record), an emitter object, and a context object (which contains CDAP metrics and logger).
+The script can then use the emitter object to emit transformed Python dictionaries.
+
+For example::
+
+   'def transform(input, emitter, context): input['count'] *= 1024; emitter.emit(input)'
+
+will scale the ``'count'`` field by 1024.
+
+**schema:** The schema of output objects. If no schema is given, it is assumed that the output
+schema is the same as the input schema.
+
+Example
+-------
+::
+
```

Comment:
One thing to note is that the script in this example is formatted to be easily readable. The actual would actually all be on one line, with escaped newlines:

```
"script": "def transform(input, emitter, context):\n    tax = input['subtotal'] * 0.0975\n    if (tax > 1000.0):\n        context.getMetrics().count('tax.above.1000', 1)\n    if (tax < 0.0):\n        context.getLogger().info('Received record with negative subtotal')\n    emitter.emit({\n        'subtotal': input['subtotal'],\n        'tax': tax,\n        'total': input['subtotal'] + tax\n    })"
```